src/modules/layout/hocs/withLayout.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+
+import React from 'react'
+import type { ComponentType } from 'react'
+import style from './Layout.css'
+
+/**
+ * The standard Layout, used for any view in this app as a container.
+ * If a footer is supplied and there's not enough content (in header and children) to fill the viewbox, the footer will
+ * always stick to the bottom of the viewbox.
+ */
+const withLayout = (
+  Header: ComponentType<{ onStickyTopChanged: number => void }>,
+  Toolbar: ?ComponentType<{}>,
+  Footer: ComponentType<{}>
```

Comment:
Not sure what you mean by that, could you add a demo usage example? 🤔

Reply:
So we could add a
* withHeader() which adds a header
* withFooter() which adds a footer
* withToolbar() which adds a toolbar

~~Not sure if that is possible but the withLocationLayout call should be documented as it's not easy to see what the first parameter does because of the recompose library. I'm not sure if you could see it easily without it.~~

Reply:
But the layout is important to stick these three things together with the correct styling...
Your idea would be
```jsx
const CategoriesPage = ....
export default compose(
  withToolbar(CategoriesToolbar),
  withHeader(LocationHeader),
  withFooter(LocationFooter),
  withFetcher(...),
  ...
)
```
? I don't know how we could even implement this...

Reply:
So if those three depend on each other then ofc leave it as one Layout. But it would be cool if they wouldn't. This would remove a lot of if-else-then code. But we don't need to do this,